master/internal/db/postgres_users.go

```diff
@@ -72,6 +76,76 @@ WHERE user_sessions.id=$1`, &user, session.ID); errors.Cause(err) == ErrNotFound
 	return &user, &session, nil
 }
 
+type externalToken struct {
+	*jwt.StandardClaims
+	Email string
+}
+
+// UserByExternalToken returns a user session derived from an external authentication token.
+func (db *PgDB) UserByExternalToken(tokenText string, tokenKey string) (*model.User,
+	*model.UserSession, error) {
+	token, err := jwt.ParseWithClaims(tokenText, &externalToken{},
+		func(token *jwt.Token) (interface{}, error) {
+			var publicKey rsa.PublicKey
+			err := json.Unmarshal([]byte(tokenKey), &publicKey)
+			if err != nil {
+				log.Errorf("error parsing JWT key: %s", err.Error())
+				return nil, err
+			}
+			return &publicKey, nil
+		},
+	)
+	if err != nil {
+		return nil, nil, err
+	}
+	claims := token.Claims.(*externalToken)
+
+	// Access control logic can be applied here
+
+	tx, err := db.sql.Beginx()
+	defer func() {
+		if tx == nil {
+			return
+		}
+
+		if rErr := tx.Rollback(); rErr != nil {
+			log.Errorf("error during rollback: %v", rErr)
+		}
+	}()
+	if err != nil {
+		return nil, nil, errors.WithStack(err)
+	}
+	user, err := db.UserByUsername(claims.Email)
```

Comment:
when access control comes in to play, this will still happen auto-magically but the user will have nil permissions right? so it'll be inconsequential.

Reply:
When access control comes into play, something else will determine is Admin / Active are true or false when the user is created / or will have already updated the database.

Reply:
cool, yeah, makes sense to me.